sympy/simplify/simplify.py

```diff
@@ -610,6 +615,17 @@ def shorter(*choices):
     if measure(expr) > ratio*measure(original_expr):
         expr = original_expr
 
+    if rewrite:
+        rewritables = set()
+        for sub in preorder_traversal(original_expr):
+            rewritables.update([attr[17:] for attr in sub.func.__dict__.keys()
+                if attr[:17] == '_eval_rewrite_as_'])
```

Comment:
- better way of getting 'rewritables': it currently preorder-transverses all nodes and gets the name from attribute of its class starting with _eval_rewrite_as_.